hello.go

```diff
@@ -21,50 +21,64 @@ import (
 	"log"
 	"net/http"
 	"os"
+	"regexp"
 )
 
 type Data struct {
 	Service      string
 	Revision     string
 	Project      string
-	ProjectFound bool
+	Region       string
 }
 
 func main() {
 	tmpl := template.Must(template.ParseFiles("index.html"))
 
 	// Get project ID from metadata server
-	project := "???"
-	projectFound := false
+	project := ""
 	client := &http.Client{}
 	req, _ := http.NewRequest("GET", "http://metadata.google.internal/computeMetadata/v1/project/project-id", nil)
 	req.Header.Set("Metadata-Flavor", "Google")
 	res, err := client.Do(req)
 	if err == nil {
 		defer res.Body.Close()
-		responseBody, err := ioutil.ReadAll(res.Body)
-		if err != nil {
-			log.Fatal(err)
+		if res.StatusCode == 200 {
+			responseBody, err := ioutil.ReadAll(res.Body)
+			if err != nil {
+				log.Fatal(err)
+			}
+			project = string(responseBody)
 		}
-		project = string(responseBody)
-		projectFound = true
 	}
 
-	service := os.Getenv("K_SERVICE")
-	if service == "" {
-		service = "???"
+	// Get region from metadata server
+	region := ""
+	req, _ = http.NewRequest("GET", "http://metadata.google.internal/computeMetadata/v1/instance/region", nil)
+	req.Header.Set("Metadata-Flavor", "Google")
+	res, err = client.Do(req)
+	if err == nil {
+		defer res.Body.Close()
+		if res.StatusCode == 200 {
+			responseBody, err := ioutil.ReadAll(res.Body)
+			if err != nil {
+				log.Fatal(err)
+			}
+			re := regexp.MustCompile(`regions\/(.*)$`)
+			match := re.FindStringSubmatch(string(responseBody))
```

Comment:
so sorry for the latency reviewing this.

can you just use replacing to remove the parts you don't want? it'll yield in shorter code:

```
regexp.MustCompile(`projects/[^/]*/regions/`).ReplaceAllString(s, "")
```

example: https://play.golang.org/p/7LCSkpEmVgu

Reply:
also please merge at your own pace.